src/components/pages/Onboarding/InputInviteToken.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import { View } from "react-native";
+
+import {
+  Button,
+  Text,
+  TextInput,
+  IndependentPageContainer
+} from "../../shared/elements";
+import { styles } from "./styles";
+import { connect, MapStateToProps } from "react-redux";
+import { RahaState } from "../../../store";
+import { InvitationsState } from "../../../store/reducers/invitations";
+import { styles as sharedStyles } from "./styles";
+import { colors } from "react-native-elements";
+
+/**
+ * Page that requests an invite token
+ */
+
+type OwnProps = {
+  onInputInviteToken: (inviteToken: string) => any;
+  onBack: () => any;
+};
+
+type ReduxStateProps = {
+  invitationsState: InvitationsState;
```

Comment:
redux nit: I think this may be fine since it's just taking the type as a dependency and maybe everything in `InvitationsState` may really be relevant to this component, but in an ideal world components that subscribe to a redux store aren't supposed to depend on the redux store in any way besides the call to redux's `connect` function/stuff passed into it (like `mapStateToProps`, etc), so that they can be used/tested/understood independently of the redux store being available (for instance, if redux were no longer the tool we wanted to use, or if we wanted to use the same frontend code in some other context). in fact, the code they suggest puts that stuff into a separate file (called "containers") to be extra clear about it.

if this component is probably the only consumer of `InvitationsState` (and therefore extraneous state probably wouldn't get passed to it), we're not really planning on reusing this frontend code in other contexts, and this is just inheriting the shape of the state, then it's fine. 

if other stuff does end up getting into the state, then that may cause this component to experience unnecessary re-renders in the future because of changing state; at that case we'd want to enumerate directly what this is supposed to actually take, which would make the component more self-documenting as well.

Reply:
What would be a better way to use a variable from the component's state (the invitation token in the case) to check if it's in invitations? I only need to check that.